R/getSubsample.R

```diff
@@ -0,0 +1,178 @@
+#' 
+#' @return a \code{SummarizedExperiment} or \code{TreeSummarizedExperiment} object 
+#'   with subsampled data that is different from input 
+#'   \code{SummarizedExperiment} or \code{TreeSummarizedExperiment}
+#' 
+#' @author Sudarshan A. Shetty 
+#' 
+#' @name getSubsample
+#'  
+#' @examples
+#' # When samples in TSE are less than specified min_size, they will be removed.
+#' # If after subsampling features are not present in any of the samples, 
+#' # they will be removed.
+#' data("GlobalPatterns")
+#' tse <- GlobalPatterns
+#' tse.subsampled <- getSubsample(tse, min_size = 60000, name = "subsampled")
+#' tse.subsampled
+#' 
+#' dim(tse)
+#' 
+#' dim(tse.subsampled)
+#' 
+NULL
+
+#' @rdname getSubsample
+#' @export
+setGeneric("getSubsample", signature = c("x"),
+           function(x, 
+                    abund_values = "counts",
+                    min_size = min(colSums2(assay(x))),
+                    seed = runif(1, 0, .Machine$integer.max),
+                    replace = TRUE,
+                    name = "subsampled",
+                    verbose = TRUE, ...)
+             standardGeneric("getSubsample"))
+
+
+#' @importFrom SummarizedExperiment assay assay<-
+#' @importFrom DelayedMatrixStats colSums2 rowSums2
+#' @rdname getSubsample
+#' @export
+setMethod("getSubsample", signature = c(x = "SummarizedExperiment"),
+          function(x, 
+                   abund_values = "counts",
+                   min_size = min(colSums2(assay(x))),
+                   seed = runif(1, 0, .Machine$integer.max), 
+                   replace = TRUE, 
+                   name = "subsampled", 
+                   verbose = TRUE, ...){
+            # Input check
+            .check_assay_present(abund_values, x)
+            if(verbose){
+              # Print to screen this value
+              message("`set.seed(", seed, ")` was used to initialize repeatable random subsampling.")
+              message("Please record this for your records so others can reproduce. \n ... \n")
+            }
+            set.seed(seed)
+            # Make sure min_size is of length 1.
+            if(length(min_size) > 1){
+              warning("`min_size` had more than one value. ", 
+                      "Using only the first. \n ... \n")
+              min_size <- min_size[1]	
+            }
+            if(min_size <= 0){
+              stop("min_size less than or equal to zero. ", 
+                   "Need positive sample size to work.")
+            }
+            # get samples with less than min number of reads
+            if(min(colSums2(assay(x, abund_values))) < min_size){
+              rmsams <- colnames(x)[colSums2(assay(x, abund_values)) < min_size]
+              if(verbose){
+                message(length(rmsams), " samples removed ",
+                        "because they contained fewer reads than `min_size`.")
+              }
+              # remove sample(s)
+              newtse <- x[, !colnames(x) %in% rmsams]
+            }
```

Comment:
`newtse` is created inside `if `--> when there are no samples that have libraries lower than threshold, `newtse` is not created --> error in line 127

Reply:
Thank for catching this.